perfmetrics/scripts/ls_metrics/listing_benchmark.py

```diff
@@ -0,0 +1,452 @@
+
+  log.info('Testing completed. Generating output.\n')
+  return gcs_bucket_results, persistent_disk_results
+
+
+def _create_directory_structure(gcs_bucket_url, persistent_disk_url, directory_structure, create_files_in_gcs) -> int:
+  """Creates new directory structure using generate_files.py as a library.
+
+  This function creates new directory structure in persistent disk. If create_files_in_gcs
+  is True, then it also creates the same structure in GCS bucket.
+  For more info regarding how the generation of files is happening, please read the generate_files.py.
+
+  Args:
+   gcs_bucket_url: Path of the directory in GCS bucket in which to create the files.
+   persistent_disk_url: Path in the persistent disk in which to create the files in.
+   directory_structure: Protobuf of the current directory.
+   create_files_in_gcs: Bool value which is True if we have to create files in GCS bucket (similar directory
+                        strucutre not present). Otherwise it is False, means that we will not
+                        create files in GCS bucket from scratch.
+
+  Returns:
+    1 if an error is encountered while uploading files to the GCS bucket. 0 if no error is encountered.
+  """
+
+  # Create a directory in the persistent disk.
+  subprocess.call('mkdir {}'.format(persistent_disk_url), shell=True)
+
+  if directory_structure.num_files != 0:
+    file_size = int(directory_structure.file_size[:-2])
+    file_size_unit = directory_structure.file_size[-2:]
+    exit_code = generate_files.generate_files_and_upload_to_gcs_bucket(gcs_bucket_url,
+                                                                       directory_structure.num_files,
+                                                                       file_size_unit, file_size,
+                                                                       directory_structure.file_name_prefix,
+                                                                       persistent_disk_url, create_files_in_gcs,
+                                                                       'return')
+    if exit_code != 0:
+      return 1
+
+  result = 0
+  for folder in directory_structure.folders:
+    result += _create_directory_structure(gcs_bucket_url + folder.name + '/',
+                                       persistent_disk_url + folder.name + '/',
+                                       folder, create_files_in_gcs)
+
+  return int(result > 0)
+
+
+def _list_directory(path) -> (list, int):
+  """Returns the list containing path of all the contents present in the current directory and also the number of contents.
+
+  Args:
+    path: Path of the directory.
+
+  Returns:
+    A list containing path of all contents present in the input path. Also returns the number of contents.
+  """
+
+  contents = subprocess.check_output('gsutil -m ls {}'.format(path), shell=True)
+  num_contents = len(contents.decode('utf-8').split('\n')[:-1])
```

Comment:
why do you have pass num_contents as a seperate return value here. the caller get the value from len(contents_url) right.  or is some data missing from contents_url list

Reply:
still not fixed. Why do you need to return num_contents from this method. Whoever wants to know the length they will use len(contents).